### PR TITLE
Issue2238

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: src/github.com/dogenzaka/rotator/.gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
```
Comment (oxtoacart) [inline src/github.com/dogenzaka/rotator/.gitignore]: Added rotator for file rotation (thanks for the suggestion @fffw!)

File: src/github.com/getlantern/appdir/appdir.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// package appdir provides a facility for determining the system-dependent
+// paths for application resources.
+package appdir
```
Comment (oxtoacart) [inline src/github.com/getlantern/appdir/appdir.go]: appdir is a new library for determining system-dependent paths for app resources.

File: src/github.com/getlantern/flashlight/config/config.go [hunk 3]
```diff
@@ -140,7 +140,7 @@ func Update(mutate func(cfg *Config) error) error {
 func InConfigDir(filename string) string {
 	cdir := *configdir
 	if cdir == "" {
-		cdir = platformSpecificConfigDir()
+		cdir = appdir.General("Lantern")
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/config/config.go]: Switched to using appdir.

File: src/github.com/getlantern/flashlight/flashlight.go [hunk 4]
```diff
@@ -169,6 +175,28 @@ func useAllCores() {
 	runtime.GOMAXPROCS(numcores)
 }
 
+func configureLogging() *rotator.SizeRotator {
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/flashlight.go]: Here's the logging stuff.

File: src/github.com/stretchr/testify/.gitignore [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
```
Comment (oxtoacart) [inline src/github.com/stretchr/testify/.gitignore]: This fork of testify is used by rotator.

**Comment by fffw** [comment]
There's a new PR https://github.com/getlantern/flashlight-build/pull/47 to merge this branch, will close this one.

**Pull request closed without merging**